components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/util/DCRMUtilsTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.dcr.util.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.oauth.dcr.DCRMConstants;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMClientException;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMServerException;
+import org.wso2.carbon.identity.oauth.dcr.util.DCRMUtils;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+public class DCRMUtilsTest {
+
+    @DataProvider(name = "BuildRedirectUrl")
+    public Object[][] BuildRedirectUrl() {
+        return new Object[][] {{"http://example.com/", true, true},{null, false ,true}, {"", false, true},
+                {"http://example.com/", true, false},{null, false ,false},{"", false, false}};
+    }
+
+    @Test(dataProvider = "BuildRedirectUrl")
+    public void testIsRedirectionUriValid(String url, boolean response, boolean isDebugenabled) throws Exception {
+        Log mockLogFactory = mock(LogFactory.getLog(DCRMUtils.class).getClass());
+        when(mockLogFactory.isDebugEnabled()).thenReturn(isDebugenabled);
```

Comment:
Refer[ [1]](https://github.com/madurangasiriwardena/identity-inbound-auth-oauth/blob/c69f1c80a3aca09b3d2366d77407239c6142b8cd/components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuerTest.java) on mocking the `LogFactory`